Makefile

```diff
@@ -173,11 +175,11 @@ gofmt: ## Runs gofmt against all packages.
 	done
 	@echo "gofmt success"; \
 
-golangci-lint: ## Run golangci-lint on codebasis
+golangci-lint: ## Run golangci-lint on codebase
 # https://stackoverflow.com/a/677212/1027058 (check if a command exists or not)
 	@if ! [ -x "$$(command -v golangci-lint)" ]; then \
-		echo "golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions."; \
-		exit 1; \
+		echo "Installing golangci-lint"; \
+		curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(GOPATH)/bin v1.21.0; \
```

Comment:
I still don't think we should install software without the permission of the user. Would you please add the installation to the circle ci build?

Reply:
Oh you want me to do it in the circleCI setup, sure.

Reply:
So ideally, I should do all the tool installations in the `setup` job in circleCI. But the installed tool needs to persist to be used in the build job. Now only `/go/src/github.com/mattermost` and its descendants are persisted, which means I have to place the binary somewhere there. Or alternatively download the tool in the `build` job itself.

What do you suggest ?

On a related note: I suggest that the tool be installed in the docker image itself. That should make the build much faster. And we just need to update the image once in a while when a new release comes out.

Reply:
Updating the docker image requires currently some more work. Having it in the CI should be fine for now.

I'm totally fine with installing it in `build`. We only use it there.